src/main/java/tech/ypsilon/bbbot/discord/command/BirthdayCommand.java

```diff
@@ -1,390 +1,404 @@
+        if (hour < notifyTime) {
+            delay = ((notifyTime - 1) - hour) * 60 + (60 - min);
+        } else {
+            delay += (23 - hour) * 60 + (60 - min);
+        }
+
+        ScheduledExecutorService ses = Executors.newScheduledThreadPool(3);
+        ScheduledFuture<?> scheduledFuture = ses.scheduleAtFixedRate(new Runnable() {
+            @Override
+            public void run() {
+                try {
+                    notifyAllGuilds();
+                } catch (Exception e) {
+                    System.err.println("Error while notifying the birthdays :(");
+                }
+            }
+        }, delay, hours * 60, TimeUnit.MINUTES);
+
+        if (NOTIFY_ON_STARTUP) {
+            new Thread(() -> {
+                try {
+                    Thread.sleep(2000);
+                    notifyAllGuilds();
+                } catch (InterruptedException e) {
+                    e.printStackTrace();
+                }
+            }).start();
+        }
+
+        ButterBrot.LOGGER.info("[Birthday]: Registered the notification-service");
+
+    }
+
+
+    /**
+     * Notifies all registered guilds about today's birthdays.
+     */
+    public static void notifyAllGuilds() {
+        BirthdayMongoDBWrapper.getRegisteredGuildIds().forEach(guildId -> notifyGuild(DiscordController.getJDA().getGuildById(guildId)));
+    }
+
+    /**
+     * Notifies the given guild in the specified "Default channel" (set by notifyhere)
+     *
+     * @param guild
+     */
+    public static void notifyGuild(Guild guild) {
+        notifyGuild(guild,
+                guild.getTextChannelById(
+                        BirthdayMongoDBWrapper.getDefaultChannel(
+                                guild.getId())));
+    }
+
+    /**
+     * Notifies the given guild in the specified channel
+     *
+     * @param guild
+     * @param channel
+     */
+    public static void notifyGuild(Guild guild, MessageChannel channel) {
+        HashMap<Long, Date> bdays = BirthdayMongoDBWrapper.getBirthdayEntrys();
+        shoutout = false;
+
+        bdays.forEach((userId, bDay) -> {
+            if (shoutOutBday(userId, bDay, guild, channel))
+                shoutout = true;
+        });
+        ;
+
+        if (!shoutout && notifyNoBirthday) {
+            channel.sendMessage("Heute gibt es leider keine Geburtstage :(").queue(message -> {
+                message.addReaction("U+1F62F").queue();
+            });
+        }
+    }
+
+    /**
+     * Sends a message in the specified channel "channel" of the guild "guild" to tell the user "name" has birthday today.
+     *
+     * @param userId
+     * @param bday
+     * @param guild
+     * @param channel
+     * @return
+     */
+    public static boolean shoutOutBday(long userId, Date bday, Guild guild, MessageChannel channel) {
+        if (hasBirthdayToday(bday)) {
+            User user = guild.getJDA().retrieveUserById(userId).complete();
+            if (guild.isMember(user)) {
```

Comment:
This will not work. Guild#isMember only checks the member cache. Since the MemberCachePolicy was not specifically set when the Discord bot was build, JDA uses DEFAULT. This will result in only members in voice channels and the owner being cached.